app/src/ereferrals/java/org/eyeseetea/malariacare/domain/usecase/GetUrlForWebViewsUseCase.java

```diff
@@ -45,12 +54,21 @@ public void execute(int urlType, Callback callback) {
 
     @Override
     public void run() {
-        String language = getUserLanguageOrDefault(mContext);
-        String typeUrl = getTypeUrlText();
-        String url = String.format(PreferencesEReferral.getWebViewURL() + mContext.getString(
-                R.string.composed_web_view_url), typeUrl, mCredentials.getUsername(),
-                mCredentials.getPassword(), language);
-        mCallback.onGetUrl(url);
+        final String language = getUserLanguageOrDefault(mContext);
+        final String typeUrl = getTypeUrlText();
+        UIThreadExecutor mainExecutor= new UIThreadExecutor();
+        IAsyncExecutor asyncExecutor = new AsyncExecutor();
+        ISettingsRepository settingsDataSource = new SettingsDataSource(mContext);
+        GetSettingsUseCase getSettingsUseCase = new GetSettingsUseCase(mainExecutor, asyncExecutor, settingsDataSource);
```

Comment:
Here there is an use case calling another use case and this is a bad practice, please review it